src/reporter/reporter.py

```diff
@@ -147,15 +168,26 @@ def notify():
         fiware_sp = request.headers.get('fiware-servicepath', None)
     else:
         fiware_sp = None
-
-    # Send valid entities to translator
-    with translator_for(fiware_s) as trans:
-        trans.insert(payload, fiware_s, fiware_sp)
-
-    msg = 'Notification successfully processed'
+    for entity in payload:
```

Comment:
Wasn't the whole `for` loop supposed to go? Not sure why it's still here. Am I missing something?

Reply:
There might still be a little snag. What's going to happen if we have this entity:

    payload = {
        'id': '299531',
        'type': 'AirQualityObserved',
        'p': { 'type': 'string' },
        'pm10': {'type': 'string', 'value': '10', 'metadata': {}},
    }

If I'm right, then:

    _filter_empty_entities(payload) is None

whereas what we want is:

    _filter_empty_entities(payload) == payload

i.e. we should **not** filter the entity out since there's an attribute holding some data, `pm10` that is.  But here's a cheap fix, probably you can come up with something better:

    ...
    Flag = False
    for j in attrs:
        value = payload[j].get('value')
        if value:
            Flag = True
    if Flag:
    ...

If you could please add a test for the above `payload` and if it breaks see if fixing `_filter_empty_entities` the way I've suggested helps---or fix it in any other way you see fit :-)

Reply:
@c0c0n3 , I have updated the code for value in entity and in def notify() it will check for entity update and if valid entity is there it will send it to translator further.
 https://github.com/smartsdk/ngsi-timeseries-api/blob/9e09c292bafa819ff6685b4c4d5db6676666bc70/src/reporter/reporter.py#L115
https://github.com/smartsdk/ngsi-timeseries-api/blob/9e09c292bafa819ff6685b4c4d5db6676666bc70/src/reporter/reporter.py#L160

Also, added test case in test_notify.py in which it will check for empty value  and for empty string 'value' in test_no_value_for_attributes(notification)
https://github.com/smartsdk/ngsi-timeseries-api/blob/b73ace5407e0a28d2c5f1aa35712ccd4b3a76534/src/reporter/tests/test_notify.py#L318

Reply:
That's great! Like I said though, I'd feel much more confident about my ability to refactor in the future if we also had a test to cover what's going to happen if the entity has both valid and empty attributes as in:

    payload = {
        'id': '299531',
        'type': 'AirQualityObserved',
        'p': { 'type': 'string' },
        'pm10': {'type': 'string', 'value': '10', 'metadata': {}},
    }

Reply:
Hie @c0c0n3 , I have updated TestCase for Both valid entity and empty value string.
https://github.com/smartsdk/ngsi-timeseries-api/blob/6ff27a7aae6b50cc423bda4acc5723e5c9782639/src/reporter/tests/test_notify.py#L363
https://github.com/smartsdk/ngsi-timeseries-api/blob/6ff27a7aae6b50cc423bda4acc5723e5c9782639/src/reporter/tests/test_notify.py#L365

In this it will check response to validate that the entity gets inserted or not using method res_get.json().